presentation/src/main/java/org/sagebionetworks/research/presentation/perform_task/PerformTaskViewModel.java

```diff
@@ -318,6 +340,10 @@ protected void updateCurrentStep(@Nullable Step nextStep, @NonNull TaskResult ta
     void handleTaskLoad(Task task) {
         LOGGER.debug("Loaded task: {}", task);
         this.task = task;
+        this.recorderManager = new RecorderManager(this.task, this.taskRunUuid, this.getApplication(), this.recorderFactory,
+                this.recorderConfigPresentationFactory);
+        // Subscribe to the recorder results and put them in the async results.
+        this.compositeDisposable.add(Flowable.create(this.recorderManager, BackpressureStrategy.BUFFER).subscribe(this::addAsyncResult));
```

Comment:
I think if the UI goes away before the io is done, we lose the data. Add a TODO

Reply:
Is this true, I thought the purpose of a view model was to outlive lifecycle states.